nixos/modules/services/web-apps/c2FmZQ-server.nix

```diff
@@ -0,0 +1,214 @@
+      default = "";
+      description = lib.mdDoc "The name of the htdigest `FILE` to use for basic auth for some endpoints, e.g. /metrics";
+    };
+
+    maxConcurrentRequests = mkOption {
+      type = types.int;
+      default = 10;
+      description = lib.mdDoc "The maximum number of concurrent requests.";
+    };
+
+    enableWebapp = mkOption {
+      type = types.bool;
+      default = true;
+      description = lib.mdDoc "Enable Progressive Web App.";
+    };
+
+    address = mkOption {
+      internal = true;
+      type = types.str;
+      default = "${cfg.bindIP}:${toString cfg.port}";
+    };
+
+    package = mkOption {
+      internal = true;
+      type = types.package;
+      default = pkgs.c2FmZQ;
+    };
+
+    args = mkOption {
+      internal = true;
+      type = types.listOf types.str;
+      default = [
+        "-database" cfg.database
+        "-address" cfg.address
+        "-path-prefix" cfg.pathPrefix
+        "-base-url" cfg.baseUrl
+        "-redirect-404" cfg.redirect404
+        "-tlscert" cfg.tlscert
+        "-tlskey" cfg.tlskey
+        "-autocert-domain" cfg.autocertDomain
+        "-autocert-address" cfg.autocertAddress
+        (optionalString cfg.allowNewAccounts "-allow-new-accounts")
+        (optionalString cfg.autoApproveNewAccounts "-auto-approve-new-accounts")
+        "-verbose" (toString cfg.verbose)
+        (optionalString cfg.encryptMetadata "-encrypt-metadata")
+        "-passphrase-command" cfg.passphraseCommand
+        "-passphrase-file" cfg.passphraseFile
+        "-passphrase" cfg.passphrase
+        "-htdigest-file" cfg.htdigestFile
+        "-max-concurrent-requests" (toString cfg.maxConcurrentRequests)
+        (optionalString cfg.enableWebapp "-enable-webapp")
+      ];
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.c2FmZQ-server = {
+      description = "c2FmZQ is an application that can securely encrypt, store, and share files, including but not limited to pictures and videos.";
+      documentation = [ "https://github.com/c2FmZQ/c2FmZQ" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "network-online.target" ];
+
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/c2FmZQ-server ${escapeShellArgs cfg.args}";
+        AmbientCapabilities = "";
+        CapabilityBoundingSet = "";
+        DynamicUser = true;
+        IPAccounting = true;
+        IPAddressAllow = cfg.bindIP;
+        IPAddressDeny = "any";
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateIPC = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
```

Comment:
This seems incredibly paranoid. Are we really running that untrustworthy a service? Is it that security sensitive? As I understand it does not have root or any special privileges